### DISCUSSION TITLE
Question about Least Loaded loadbalancing rule

### DISCUSSION THREAD

**Discussion by ajayvtiwari** [linkerd/linkerd2#5920] [Show and tell] [answered]
Hello,

I have implemented lead-loaded load balancing algorithm but had two questions on the same

1) Can someone confirm the Least Loaded algorithm is the same as least-connection as the documentation says that the loadbalancing decision is based on least number of requests which is the same as least-connection

_Power of Two Choices (P2C): Least Loaded
This is Linkerd's default load balancer. When determining where to send a request, Linkerd picks two replicas from the load balancer pool at random, and it selects the one that is the least loaded of the two. Load is determined by the number of outstanding requests to each replica. This algorithm provides a manageable upper bound on load for individual replicas. It does so using less overhead than other algorithms with similar performance_

2) How do we test the least-loaded algorithm? What tools can I use to see how many requests at a given point of time are being processed by a POD and how do I test the load?

Regards,
Ajay

**Comment by kleimkuhler** [comment, marked as answer]
@ajayt77 Using number of requests will not be the same as number of connections. There could be a small number of very chatty clients resulting in many outstanding requests, or a large number of idle clients resulting is few outstanding requests.

For testing I'd recommend looking at one of demo apps [slow_cooker](https://github.com/BuoyantIO/slow_cooker) and changing the default settings to ones that are helpful to you. You can use any of Linkerd's observability tools to get an idea about requests and connections. In particular, check out the `linkerd diagnostics proxy-metrics` command.
